### ISSUE TITLE
Disable properties at runtime

### ISSUE THREAD

**Issue by VeNoMouS** [guessit-io/guessit#231]
Filename: Running.Wild.With.Bear.Grylls.S02E07.Michael.B.Jordan.PROPER.HDTV.x264-W4F.avi

```
>>> import guessit
>>> guessit.guess_episode_info("Running.Wild.With.Bear.Grylls.S02E07.Michael.B.Jordan.PROPER.HDTV.x264-W4F.avi",info=['filename'])
{
u'mimetype': 'video/x-msvideo', 
u'episodeNumber': 7,
u'videoCodec': u'h264',
u'container': u'avi',
u'title': u'Michael B',
u'series': u'Running Wild With Bear Grylls (JO)',
u'format': u'HDTV',
u'releaseGroup': u'W4F',
u'other': [u'Proper'],
u'season': 2,
u'properCount': 1,
'country': <Country [JO]>,
u'type': u'episode'
}
>>> 
```

@wackou you know a work around for this?

**Comment by VeNoMouS** [comment]
NM... Open ticket #229 helped me out :)

```
>>> guessit.guess_episode_info("Running.Wild.With.Bear.Grylls.S02E07.Michael.B.Jordan.PROPER.HDTV.x264-W4F.avi",info=['filename'],options={'disabled_transformers':'GuessCountry'})
{
u'mimetype': 'video/x-msvideo', 
u'episodeNumber': 7,
u'videoCodec': u'h264',
u'container': u'avi',
u'title': u'Michael B Jordan',
u'series': u'Running Wild With Bear Grylls',
u'format': u'HDTV',
u'releaseGroup': u'W4F',
u'other': [u'Proper'],
u'season': 2, 
u'properCount': 1,
u'type': u'episode'
}
``

**Comment by Toilal** [comment]
I'll exclude Jordan word from contries detection.

**Comment by srob650** [comment]
Did this ever get fixed?  I'm having the same issue with a different filename

**Comment by VeNoMouS** [comment]
@srob650 i posted the fix in my previous comment.. guess you didn't read... disable ```guessCountry``` to fix the issue..

**Comment by srob650** [comment]
Ah, I did gloss over that. Thanks!

**Comment by srob650** [comment]
Hmm, looks like this was actually removed in v2, which I am using

**Comment by VeNoMouS** [comment]
Yup would appear your correct, according to docs/migrations.rst
```
Options
-------
Some options have been removed.

- ``-X DISABLED_TRANSFORMERS``, ``-s, --transformers``

  There's no transformer anymore.
```

**Comment by VeNoMouS** [comment]
```
API
----
``guess_video_info``, ``guess_movie_info`` and ``guess_episode_info`` have been removed in favor of a unique function guessit
```

UGGH!!

**Comment by VeNoMouS** [comment]
Ok @srob650  if you define some bullshit country it overrites the bs country detection..

ie

```
>>> guessit.guessit(u"Running.Wild.With.Bear.Grylls.S02E07.Michael.B.Jordan.PROPER.HDTV.x264-W4F.avi",options={'allowed_countries':'.'})

MatchesDict([(u'title', u'Running Wild With Bear Grylls'), (u'season', 2), (u'episode', 7), (u'episode_title', u'Michael B Jordan'), (u'other', u'Proper'), (u'proper_count', 1), (u'format', u'HDTV'), (u'video_codec', u'h264'), (u'release_group', u'W4F'), (u'container', u'avi'), (u'mimetype', 'video/x-msvideo'), (u'type', u'episode')])
```

**Comment by srob650** [comment]
This will definitely suffice for now, even though it's ugly, thank you @VeNeMouS 

I definitely disagree with the concept of modifying the series name in any way, and would vote to not add the country to the name even if it is detected. This is what the dictionary keys are for, let the user decide to do what they want with the country of detected.

**Comment by Toilal** [comment]
I'm agree with you @srob650, I think i'll just remove the appending of country in the name before releasing final 2.0.

@VeNoMouS : About the removal of `disabled-transformers` option, maybe I could add a `disabled-properties` option that would totally disable some of the properties from their names (ie:  `--disabled-properties country --disabled-properties language`).

About the removal of various api methods ``guess_video_info``, ``guess_movie_info`` and ``guess_episode_info``, it's only a matter of simplifying the api entry points. 

Now there's a single method, and filetype is given into options. 

It was already the case for guessit 1.x, [those 3 methods were only shortcuts](https://github.com/guessit-io/guessit/blob/1.x/guessit/__init__.py#L361-L370) to `guess_file_info` with `type` defined in keyword argument.

It's far more [clear and consistent now in guessit 2](https://github.com/guessit-io/guessit/blob/master/guessit/api.py#L58-L71), where all options are given to a single `guessit` function through a dict or [shlex](https://docs.python.org/3/library/shlex.html) string.

But you should really consider migrating to guessit 2, because it's already really stable, and will hit final release soon (January 2016).

**Comment by srob650** [comment]
@Toilal Awesome, glad to hear you agree!  I've been using v2 for a bit now and it has been great except for this issue, so thanks for the great work!

As far as disabling properties, it would still be nice to have the option, especially if disabling them gives some performance increase (which I naively assume it would?).  My use case involves iterating over tens of thousands of filenames so any little performance increase helps!

**Comment by VeNoMouS** [comment]
My only issues are.. I have to re-code... :P backwards compatibility would've been nice... and perhaps a blurb about the the removed functions would have been nice in the migration document rather than "they are just gone" , i don't really care about transfomers... I only use it to disable the country code lookup, I was in the same boat as @srob650 i had to read your code in order to workout how to disable them due to the lack of documentation.

Still love your work man :) don't get me wrong about that, just the migration path from 1.x to 2.x could have been documented / exampled a bit better

**Comment by Toilal** [comment]
Yes, i'm sorry you have to recode, but it should be quite simple. Here's a migration Pull Request i've written for FlexGet : https://github.com/Flexget/Flexget/pull/646/files. Not the main changes are renamed properties from camelCase to underscore_case, and api method name change.

I have decided to introduce those breaking changes in this new major version, because there were some design issues in previous version that were impossible to fix without those changes.

Also, for newcomers wanting to embed guessit 2 from scracth, it's really easier and more consistent.

But I agree there's something missing for people using `-X --disabled-transformers` option. I'll try to implement the `--disabled-property` as soon as possible and before the 2.0 final release, so there'll be then something else than "transformers are just gone" :)

**Comment by Toilal** [comment]
Also, almost all unit tests cases have been kept, so results should be about the same from guessit 1.x to guessit 2.x.

**Comment by VeNoMouS** [comment]
Thanks @Toilal

**Comment by srob650** [comment]
Did some testing with the new commit, seems to work great thank you @Toilal!  As far as I'm concerned this issues is closed :)

**Comment by VeNoMouS** [comment]
Just read the commit... nice :)

**Comment by Toilal** [comment]
I still have to implement property ignore and i'll close the issue.

**Comment by srob650** [comment]
Sounds good

**Comment by srob650** [comment]
Having trouble with a related query to this thread again.

Show: Ninja Warrior UK
(UK is actually part of the title)
```python
>>> guessit(u'ninja warrior uk s01e01')
>>> MatchesDict([(u'title', u'ninja warrior'), (u'country', <Country [GB]>), (u'season', 1), (u'episode', 1), (u'type', u'episode')])
```

Is there a way for me to get the correct result of (u'title', u'ninja warrior uk')  or am I screwed on this one?

**Comment by VeNoMouS** [comment]
yeaaaa, unsure of that one, @Toilal  might have to put in an exception.

**Comment by srob650** [comment]
Yeah, not a huge deal, just curious if there was a method for this I was unaware of.  Seems like it might be possible to add an option to disable country detection that keeps things that look like countries in the title, but probably pretty low on the list (I can't think of any other examples like this).

**Comment by Toilal** [comment]
From wikipedia, "Ninja Warrior UK is the British version of the Japanese assault course game show Sasuke". So from objective point of view, result you have is good.

You can postprocess results to append country to the title value ...

**Comment by srob650** [comment]
I would disagree, "UK" is clearly part of the actual official title, not just a descriptor of where the show airs.  In any case, it's all good, this is a far edge case :)

**Comment by ratoaq2** [comment]
Fixed by #513
